File: myTrial-api/src/main/java/com/cmpe138/mytrial/service/ResearcherService.java
// SJSU CMPE 138 Fall 2021 TEAM1
package com.cmpe138.mytrial.service;

import java.util.ArrayList;
import java.util.List;

import com.cmpe138.mytrial.model.Patient;
import com.cmpe138.mytrial.model.Researcher;

public interface ResearcherService {

	List<Researcher> getAll();

	Researcher getResearchertById(String researcher_id);
	
	Researcher getResearcherByUsernamePassword(String username, String password);
	
	List<String> getResearcherByTrialId(String trial_id);
	
	void createResearcher(String r_name, String r_username, String r_password, List<String> organizations, List<String> disease_areas);
}

File: mytrial-fe/src/service/ResearcherService.js
// SJSU CMPE 138 Fall 2021 TEAM1
import axios from 'axios';
import MYTRIAL_API from '../settings/index'
const RESEARCHER_API_BASE_URL = `${MYTRIAL_API}/researcher`;

class ResearcherService {

    getResearchers() {
        return axios.get(RESEARCHER_API_BASE_URL);
    }

    createResearcher(researcher) {
        return axios.post(RESEARCHER_API_BASE_URL, researcher);
    }

    getResearcherById(researcherId) {
        return axios.get(RESEARCHER_API_BASE_URL + '/' + researcherId);
    }

    updateResearcher(researcher, researcherId) {
        return axios.put(RESEARCHER_API_BASE_URL + '/' + researcherId, researcher);
    }

    deleteResearcher(researcherId) {
        return axios.delete(RESEARCHER_API_BASE_URL + '/' + researcherId);
    }
}

export default new ResearcherService()
